parachain/runtime/tests/test_nomination.rs

```diff
@@ -0,0 +1,354 @@
+        let issuance_capacity_before_withdrawal_request =
+            VaultRegistryPallet::get_issuable_tokens_from_vault(account_of(VAULT)).unwrap();
+        assert_ok!(request_nominator_collateral_withdrawal(USER, VAULT, DEFAULT_NOMINATION));
+        let issuance_capacity_after_withdrawal_request =
+            VaultRegistryPallet::get_issuable_tokens_from_vault(account_of(VAULT)).unwrap();
+        assert_eq!(issuance_capacity_before_withdrawal_request, 623333);
+        assert_eq!(issuance_capacity_after_withdrawal_request, 556666);
+    });
+}
+
+#[test]
+fn integration_test_operator_cannot_withdraw_directly() {
+    test_with_nomination_enabled_and_operator_registered(|| {
+        assert_nominate_collateral(USER, VAULT, DEFAULT_NOMINATION);
+        assert_noop!(
+            Call::VaultRegistry(VaultRegistryCall::withdraw_collateral(100000000))
+                .dispatch(origin_of(account_of(VAULT))),
+            VaultRegistryError::NominationOperatorCannotWithdrawDirectly
+        );
+    });
+}
+
+#[test]
+fn integration_test_nominator_withdrawal_below_collateralization_threshold_fails() {
+    test_with_nomination_enabled(|| {
+        assert_ok!(
+            Call::VaultRegistry(VaultRegistryCall::withdraw_collateral(800000)).dispatch(origin_of(account_of(VAULT)))
+        );
+        assert_register_operator(VAULT);
+        assert_nominate_collateral(USER, VAULT, DEFAULT_NOMINATION);
+        assert_ok!(ExchangeRateOraclePallet::_set_exchange_rate(
+            FixedU128::checked_from_integer(3).unwrap()
+        ));
+        assert_noop!(
+            request_nominator_collateral_withdrawal(USER, VAULT, 10),
+            VaultRegistryError::InsufficientCollateral
+        );
+    });
+}
+
+#[test]
+fn integration_test_operator_withdrawal_can_force_refunds_nominators() {
+    test_with_nomination_enabled(|| {
+        assert_ok!(
+            Call::VaultRegistry(VaultRegistryCall::withdraw_collateral(800000)).dispatch(origin_of(account_of(VAULT)))
+        );
+        assert_register_operator(VAULT);
+        assert_nominate_collateral(USER, VAULT, DEFAULT_NOMINATION);
+        let user_collateral_balance_before_force_refund = CollateralPallet::get_balance_from_account(&account_of(USER));
+        let amount_to_withdraw = 10000;
+        assert_ok!(request_operator_collateral_withdrawal(VAULT, amount_to_withdraw));
+        let user_collateral_balance_after_force_refund = CollateralPallet::get_balance_from_account(&account_of(USER));
+        let max_nomination_ratio = NominationPallet::get_max_nomination_ratio();
+        let expected_force_liquidation = FeePallet::dot_for(amount_to_withdraw, max_nomination_ratio).unwrap();
+        assert_eq!(
+            user_collateral_balance_before_force_refund + expected_force_liquidation,
+            user_collateral_balance_after_force_refund
+        );
+    });
+}
+
+#[test]
+fn integration_test_liquidating_operators_for_low_collateralization_also_liquidates_nominators() {
```

Comment:
In pretty much all integration tests that check balances, I would prefer to use the `ParachainState` struct, so that we are sure we are sure that any amounts that we don't check remain unchanged. See `integration_test_issue_polka_btc_execute_bookkeeping` for an example. You'll need to edit `ParachainState` to include a nominator